telescope/selector_test.py

```diff
@@ -40,8 +40,18 @@ def assertSelectorMatches(self, selector_expected, selector_actual):
     self.assertEqual(selector_expected.ip_translation_spec.strategy_name,
                      selector_actual.ip_translation_spec.strategy_name)
     self.assertDictEqual(selector_expected.ip_translation_spec.params, selector_actual.ip_translation_spec.params)
-    self.assertEqual(selector_expected.site, selector_actual.site)
-    self.assertEqual(selector_expected.client_provider, selector_actual.client_provider)
+
+    if hasattr(selector_expected, 'site') or \
+            selector_actual.site != None:
+        self.assertEqual(selector_expected.site, selector_actual.site)
```

Comment:
Do we need to do the `if`? Won't every `Selector` have a `site` attribute by virtue of the `__init__` method? Given that, it seems like we can do the `assertEqual` without any pre-check.